### PR TITLE
Fix customized paper generation error when not logged in

### LINKED ISSUE #34: Customized paper generation when not logged in gives error 500

### PR DESCRIPTION
**Pull request by KIRA009**
This fixes #34 

As of this commit, the error that occurred while generating customized
paper without logging in is fixed. The GeneratedCustomizedPaper takes a
mentor username as a filter parameter, whereas, when not logged in, it
was being fed AnonymousUser object, which was breaking the code.

This issue was fixed by checking if the user is authenticated right off
the bat, thereby not waiting for the code to reach some stage where it
could break, since anyways the user should not be allowed to generate a
customised paper without logging in. The user is then prompted to log
in, following which he is redirected to the required url from the 'next'
query parameter, for which slight alteration is also done in login_view
function

Also, this commit fixes(probably) some weird behaviour, which is,
in get_customize_paper, when chapters list is evaluated, because of the
nature of split function, an empty string is converted to a list with an
element which is an empty string as well. Running map on this list with
int function means that int('') is run, which raises an Exception. This is
fixed naively by checking for that exact condition, and thereby avoid
the exception from occurring.

### CODE REVIEW

**Comment by ishanisri** [comment]
Please Pull master, rebase and push. 
Also, can you check that when you upload the data for weak students, are the chapters being displayed or not in the field to select "chapters to exclude"?

File: searchapp/views.py [hunk 1]
```diff
@@ -402,12 +405,18 @@ def get_generic_paper(request):
...
 def get_customize_paper(request):
+    if not request.user.is_authenticated:
+        return JsonResponse({"error": "To access customised paper, you need to be logged in"}, status=401)
     if request.method == 'POST':
         subject = request.POST['subject']
         # filter returns a query set which is converted to a list and then the first element is picked up.
         subject_name = list(Subject.objects.filter(
             id=subject).values_list('subject_name', flat=True))[0]
-        chapters = map(int, request.POST.getlist('chapters[]')[0].split(','))
+        chapters = request.POST.getlist('chapters[]')[0].split(',')
```
Comment (PaiAkshay998) [inline searchapp/views.py]: @KIRA009 why was this part changed?

File: searchapp/views.py [hunk 2]
```diff
@@ -402,12 +405,18 @@ def get_generic_paper(request):
...
+        if chapters[0]:
```
Comment (PaiAkshay998) [inline searchapp/views.py]: This will throw an IndexError if chapters is empty.

File: searchapp/views.py (hunk 1 above)
Comment (KIRA009) [reply searchapp/views.py]: I mentioned the reason in the commit message
> when chapters list is evaluated, because of the
nature of split function, an empty string is converted to a list with an
element which is an empty string as well. Running map on this list with
int function means that int('') is run, which raises an Exception.

File: searchapp/views.py (hunk 2 above)
Comment (KIRA009) [reply searchapp/views.py]: If `chapters[][0]` is empty, using `.split(',')` on it will return a list with one element, which is an empty string. So, this line basically checks that condition itself.

```
>>> x = ''.split(',')
>>> x[0]
''
```
